models/migrations/v1_20/v255.go

```diff
@@ -0,0 +1,21 @@
+// Copyright 2023 The Gitea Authors. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+package v1_20 //nolint
+
+import (
+	"code.gitea.io/gitea/modules/timeutil"
+
+	"xorm.io/xorm"
+)
+
+func AddArchivedUnixToRepository(x *xorm.Engine) error {
+	type Repository struct {
+		ArchivedUnix timeutil.TimeStamp `xorm:"Default 0"`
+	}
+
+	x.Sync(new(Repository))
+
+	_, err := x.Exec("UPDATE repository SET archived_unix = 0")
```

Comment:
I don't quite understand the purpose of this statement.
That should already be done through the `DEFAULT`.

Or do you perhaps want to set it to `now` for all archived repos that don't have an archival time yet?

Reply:
This was [posted by lunny](https://github.com/go-gitea/gitea/pull/22664#discussion_r1171375981l): I also don't see why we need this, but perhaps some databases have Problem without running Update.

> Or do you perhaps want to set it to `now` for all archived repos that don't have an archival time yet?

That would result in a lot of incorrect data and confusion (Why was this Repo archived in 2023, but the last commit was in 2019? What happened in the meantime? Maybe some commits gets removed with a force push?), so I don't think this is a good idea. If we don't know the Date, we should not show a Date instead of a random wrong Date.

Reply:
It should be `SET archived_unix = updatedat_unix`

Reply:
`updated_unix` is 0 when the date is unknown or the Repo is not archived. It can never be NULL, so we don't need zhe where here.

Reply:
Actually, @lafriks' SQL is false, it needs to be
```sql
UPDATE repository SET archived_unix = updated_unix WHERE is_archived = 1 AND archived_unix = 0;
```
as you otherwise overwrite existing ones.

Reply:
If the Migration is run, the `archived_unix` column will be added, what means this Column has no other Values than 0.

Reply:
No, you cannot assume that.
There are many possible reasons why you would need to downgrade and rerun the migration again, beginning with local deployment as a maintainer to test out features, or a broken upgrade.
Asymmetric migrations should be avoided.

Reply:
Should we really set `archived_unix` as the latest Update Date or should we left it with 0? Is it guaranteed that the latest update date is also the archive date? Does archiving a Repo also change `updated_unix`?

Reply:
`updated_unix` is always updated when the `repository` entry is saved.
In our example, archival sets `is_archived = true`, thus the entry is updated and `updated_unix` with it.
So yes, it's guaranteed.

Reply:
`updated_unix` is always updated when the `repository` entry is saved.

If I would have known that I wouldn't have added a new column, but now we have to Column, so we could stay with it. The Command is now updated.

Reply:
This means that the Date can be wrong. I'm more in favour of "If we don't know the exact Date, we should not assume a Date", but if the most of you wants this, i can stay with it.

Reply:
Hmm, I agree with "This means that the Date can be wrong" now, so I am neutral for this change. So LGTM.